### PR TITLE
Bug# 6733: refactor: cleanup and simplify logic flow

### PR DESCRIPTION
**Pull request by garvankeeley**
I found this code hard to follow in a few places, I am very familiar with this code so it was easy enough to refactor into something I can better understand.

### CODE REVIEW

File: RustFxA/FxAWebViewController.swift [hunk 1]
```diff
@@ -31,21 +31,13 @@ class FxAWebViewController: UIViewController, WKNavigationDelegate {
...
      */
     init(pageType: FxAPageType, profile: Profile, dismissalStyle: DismissType, deepLinkParams: FxALaunchParams?) {
-        self.viewModel = FxAWebViewModel(
-            pageType: pageType,
-            profile: profile,
-            deepLinkParams: deepLinkParams,
-            firefoxAccounts: RustFirefoxAccounts.shared,
-            leanPlumClient: LeanPlumClient.shared
-        )
+        self.viewModel = FxAWebViewModel(pageType: pageType, profile: profile, deepLinkParams: deepLinkParams)
```
Comment (garvankeeley) [inline RustFxA/FxAWebViewController.swift:34]: Notes: `profile` has fxa as a property, leanplum is used as a shared by-convention and isn't passed around.

File: RustFxA/FxAWebViewController.swift [hunk 2]
```diff
@@ -73,11 +65,8 @@ class FxAWebViewController: UIViewController, WKNavigationDelegate {
         webView.navigationDelegate = self
         view = webView
         
-        viewModel.authenticate()
-        
-        viewModel.onEmittingNewState = { [weak self] output in
-            let (request, method) = output
-            if let method = method {
+        viewModel.setupFirstPage { [weak self] (request, telemetryEventMethod) in
```
Comment (garvankeeley) [inline RustFxA/FxAWebViewController.swift:68]: Notes: `authenticate` isn't accurate, `setupFirstPage` is, and a completion handler is much simpler

File: RustFxA/FxAWebViewController.swift [hunk 3]
```diff
@@ -86,10 +75,6 @@ class FxAWebViewController: UIViewController, WKNavigationDelegate {
         viewModel.onDismissController = { [weak self] in
             self?.dismiss(animated: true)
         }
-        
-        viewModel.onWantingToExecuteJSScript = { [weak self] msg in
-            self?.webView.evaluateJavaScript(msg)
-        }
```
Comment (garvankeeley) [inline RustFxA/FxAWebViewController.swift:92]: Note: confusing to bounce back here for this one-line

File: RustFxA/FxAWebViewController.swift [hunk 4]
```diff
@@ -111,15 +96,16 @@ class FxAWebViewController: UIViewController, WKNavigationDelegate {
...
     func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
-        viewModel.parseAndExecuteSuitableRemoteCommand(basedOn: message)
+        viewModel.handle(scriptMessage: message)
     }
 }
 
 extension FxAWebViewController {
     func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
-        viewModel.dispatchLongPressCommand()
+        let hideLongpress = "document.body.style.webkitTouchCallout='none';"
```
Comment (garvankeeley) [inline RustFxA/FxAWebViewController.swift:106]: There is no model logic inherent in doing this, do it here.

File: RustFxA/FxAWebViewController.swift [hunk 5]
```diff
@@ -144,7 +130,7 @@ extension FxAWebViewController: WKUIDelegate {
         helpBrowser = wv
         helpBrowser?.navigationDelegate = self
 
-        navigationItem.leftBarButtonItem = UIBarButtonItem(title: FxAWebViewModel.BackTitle, style: .plain, target: self, action: #selector(closeHelpBrowser))
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: Strings.BackTitle, style: .plain, target: self, action: #selector(closeHelpBrowser))
```
Comment (garvankeeley) [inline RustFxA/FxAWebViewController.swift:133]: nit: capitalization was wrong on FxAWebViewModel.BackTitle, but also the helpviewer logic is already here, no reason to jump there for this.

File: RustFxA/FxAWebViewModel.swift [hunk 6]
```diff
@@ -9,17 +9,6 @@ import MozillaAppServices
...
-extension URL {
-    var toRequest: URLRequest {
-        return URLRequest(url: self)
-    }
-    /// Figuring out if two urls are the same based on scheme, host and path.
-    func isEquivilent(to url: URL) -> Bool {
-        return self.scheme == url.scheme && self.host == url.host && self.path == url.path
-    }
-}
-
```
Comment (garvankeeley) [inline RustFxA/FxAWebViewModel.swift:22]: Note on URL extensions:
- already extensive, disagree with adding more for single-use 
- we have an URL extension class, how would someone know to look here?
- spelling is wrong
- equivalency is not something we want to add broadly, is a complicated topic, there is a specific form of check used here we can inline

File: RustFxA/FxAWebViewModel.swift [hunk 7]
```diff
@@ -39,90 +28,60 @@ fileprivate enum RemoteCommand: String {
     case profileChanged = "profile:change"
 }
 
-/// Represent the state of FxAWebViewModel
-typealias FxASTATE = (URLRequest, UnifiedTelemetry.EventMethod?)
```
Comment (garvankeeley) [inline RustFxA/FxAWebViewModel.swift:43]: nit: the capitalization on this is wrong, watch for this in reviews.

**Comment by nbhasin2** [comment]
Hmm its odd that buddybuild succeeded but its still showing waiting for status to be reported 🤔
